app/screens/send-bitcoin-screen/payment-details/utils.ts

```diff
@@ -0,0 +1,127 @@
+import { WalletCurrency } from "@app/graphql/generated"
+import {
+  AmountInvalidReason,
+  AmountStatus,
+  LimitType,
+  PaymentDetail,
+} from "./index.types"
+import {
+  BtcMoneyAmount,
+  UsdMoneyAmount,
+  greaterThan,
+  isNonZeroMoneyAmount,
+  moneyAmountIsCurrencyType,
+  toUsdMoneyAmount,
+} from "@app/types/amounts"
+import { PaymentType } from "@galoymoney/client/dist/parsing-v2"
+
+export const isValidAmount = ({
+  paymentDetail,
+  usdWalletAmount,
+  btcWalletAmount,
+  withdrawalLimits,
+  intraledgerLimits,
+}: {
+  paymentDetail?: PaymentDetail<WalletCurrency>
+  usdWalletAmount: UsdMoneyAmount
+  btcWalletAmount: BtcMoneyAmount
+  withdrawalLimits?: readonly {
+    readonly __typename: "OneDayAccountLimit"
+    readonly totalLimit: number
+    readonly remainingLimit?: number | null
+    readonly interval?: number | null
+  }[]
+  intraledgerLimits?: readonly {
+    readonly __typename: "OneDayAccountLimit"
+    readonly totalLimit: number
+    readonly remainingLimit?: number | null
+    readonly interval?: number | null
+  }[]
+}): AmountStatus => {
+  if (!paymentDetail || !isNonZeroMoneyAmount(paymentDetail.settlementAmount)) {
+    return {
+      validAmount: false,
+      invalidReason: AmountInvalidReason.NoAmount,
+    } as const
+  }
+
+  const settlementAmount = paymentDetail.settlementAmount
+  if (
+    moneyAmountIsCurrencyType(settlementAmount, WalletCurrency.Btc) &&
+    greaterThan({
+      value: settlementAmount,
+      greaterThan: btcWalletAmount,
+    })
+  ) {
+    return {
+      validAmount: false,
+      invalidReason: AmountInvalidReason.InsufficientBalance,
+      balance: btcWalletAmount,
+    }
+  }
+
+  if (
+    moneyAmountIsCurrencyType(settlementAmount, WalletCurrency.Usd) &&
+    greaterThan({
+      value: settlementAmount,
+      greaterThan: usdWalletAmount,
+    })
+  ) {
+    return {
+      validAmount: false,
+      invalidReason: AmountInvalidReason.InsufficientBalance,
+      balance: usdWalletAmount,
+    }
+  }
+
+  const usdAmount = paymentDetail.convertMoneyAmount(
+    paymentDetail.unitOfAccountAmount,
+    WalletCurrency.Usd,
+  )
+
+  for (const intraledgerLimit of intraledgerLimits || []) {
```

Comment:
could be a lot more efficient to just fetch the limit we need? 

this is a resource intensive operation in the backend.

Reply:
Do you mean just fetch intraledger or just fetch withdrawal?